fil-ir/src/info.rs

```diff
@@ -815,6 +829,7 @@ impl Reason {
                     .with_message("event's delay must be greater than the instance's borrow length")
                     .with_labels(vec![live, ev])
             }
+            Reason::Generated { reason } => reason.diag(ctx),
```

Comment:
I would add some way for the label to say that this was generated after elaboration